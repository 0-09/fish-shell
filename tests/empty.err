
####################
# Testing `0` after empty functions and blocks
