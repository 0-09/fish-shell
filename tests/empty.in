# See issue 5692

logmsg "Testing `$status` after empty functions and blocks"

function empty
end

# functions shouldn't preserve $status

false; empty; echo $status # 0
true; empty; echo $status # 0

# blocks should preserve $status

false; begin; end; echo $status # 1
true; begin; end; echo $status # 0
