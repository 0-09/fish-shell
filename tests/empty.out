
####################
# Testing `0` after empty functions and blocks
0
0
1
0
